src/main/scala/org/apache/spark/sql/execution/datasources/greenplum/GreenplumUtils.scala

```diff
@@ -97,9 +103,28 @@ object GreenplumUtils extends Logging {
       }
     }
 
-    df.foreachPartition { rows =>
+    var finalDf = df
+    if (isAppend) {
+      // TODO: For the case, which should append data to table, is there more efficient method?
+      finalDf = df.coalesce(1)
+    }
+
+    val partNum = finalDf.rdd.getNumPartitions
+    val accumulator = finalDf.sparkSession.sparkContext.longAccumulator("copySuccess")
+    val randomString = UUID.randomUUID().toString.flatMap {
+      case '-' => ""
+      case c => s"$c"
+    }
+
+    var tempTable = options.table + randomString
```

Comment:
What about add some human readable suffix or prefix such as "__sparkGpTmp__" in case we have to delete those tables manually

Reply:
done